<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>416</width>
    <height>638</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>.AppleSystemUIFont</family>
   </font>
  </property>
  <property name="windowTitle">
   <string>Polynomial Calculator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,0,0,0,0,0">
    <item>
     <widget class="QLabel" name="labelHint">
      <property name="text">
       <string>多项式与表达式求值计算器</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLineEdit" name="display">
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
      <property name="placeholderText">
       <string>显示结果或错误信息</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="layoutPolys">
      <item>
       <widget class="QLineEdit" name="lineA">
        <property name="text">
         <string>A = </string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineB">
        <property name="text">
         <string>B = </string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="layoutOps">
      <item>
       <widget class="QPushButton" name="btnAdd">
        <property name="text">
         <string>A + B</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnSub">
        <property name="text">
         <string>A - B</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnMulPoly">
        <property name="text">
         <string>A × B</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnDerivA">
        <property name="text">
         <string>d(A)/dx</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnDerivB">
        <property name="text">
         <string>d(B)/dx</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="layoutEval">
      <item>
       <widget class="QLabel" name="labelEval">
        <property name="text">
         <string>Eval A at x:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="xInput">
        <property name="placeholderText">
         <string>x</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnEvalA">
        <property name="text">
         <string>Go</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="layoutEdit">
      <item>
       <widget class="QPushButton" name="btnEditA">
        <property name="text">
         <string>Edit A</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnSaveA">
        <property name="text">
         <string>Save A</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnEditB">
        <property name="text">
         <string>Edit B</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnSaveB">
        <property name="text">
         <string>Save B</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QGridLayout" name="layoutKeys">
      <item row="0" column="0">
       <widget class="QPushButton" name="btn7">
        <property name="text">
         <string>7</string>
        </property>
       </widget>
      </item>
      <item row="2" column="4">
       <widget class="QPushButton" name="btnMinus">
        <property name="text">
         <string>-</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QPushButton" name="btn1">
        <property name="text">
         <string>1</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QPushButton" name="btn4">
        <property name="text">
         <string>4</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QPushButton" name="btn3">
        <property name="text">
         <string>3</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="btn5">
        <property name="text">
         <string>5</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="btn8">
        <property name="text">
         <string>8</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QPushButton" name="btnBack">
        <property name="text">
         <string>⌫</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QPushButton" name="btn2">
        <property name="text">
         <string>2</string>
        </property>
       </widget>
      </item>
      <item row="3" column="4">
       <widget class="QPushButton" name="btnMul">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="btn6">
        <property name="text">
         <string>6</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QPushButton" name="btnClear">
        <property name="text">
         <string>C</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="btn9">
        <property name="text">
         <string>9</string>
        </property>
       </widget>
      </item>
      <item row="4" column="4">
       <widget class="QPushButton" name="btnPow">
        <property name="text">
         <string>^</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QPushButton" name="btn0">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="5" column="2">
       <widget class="QPushButton" name="btnDot">
        <property name="text">
         <string>.</string>
        </property>
       </widget>
      </item>
      <item row="5" column="4">
       <widget class="QPushButton" name="btnDivide">
        <property name="text">
         <string>/</string>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="btnPlus">
        <property name="text">
         <string>+</string>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QPushButton" name="btnLParen">
        <property name="text">
         <string>(</string>
        </property>
       </widget>
      </item>
      <item row="2" column="5">
       <widget class="QPushButton" name="btnRParen">
        <property name="text">
         <string>)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="5">
       <widget class="QPushButton" name="btnX">
        <property name="text">
         <string>x</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="layoutEvalExpression">
      <item>
       <widget class="QPushButton" name="btnEvalExpression">
        <property name="text">
         <string>Evaluate Expression</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
